### PR TITLE
Keep track of the character to glyph mapping in font_renderer.js, to prevent errors when different characters point to the same glyph (issue 7101)

### LINKED ISSUE #7101: PDF file with "É" char :
Attached file can not be loaded.
Uncaught Error: Requesting object that isn't resolved yet g_font_2_path_E
[EBug.pdf](https://github.com/mozilla/pdf.js/files/184631/EBug.pdf)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
~~_I've not yet had time to create a reduced test-case, and I'm not sure if we'll be able to test this using only an appropriate text rendering mode (i.e. `4` - `7`), or if `disableFontFace = true` is actually necessary._~~

Fixes #7101.

### CODE REVIEW

File: src/core/font_renderer.js [hunk 1]
```diff
@@ -656,7 +669,8 @@ var FontRendererFactory = (function FontRendererFactoryClosure() {
     this.glyphs = glyphs;
     this.cmap = cmap;
 
-    this.compiledGlyphs = [];
+    this.compiledGlyphs = Object.create(null);
+    this.compiledCharCodeToGlyphId = Object.create(null);
```
Comment (Snuffleupagus) [inline src/core/font_renderer.js]: Looking through this once more, I'm assuming that these two lines could actually be removed given the `CompiledFont.call(this, fontMatrix);` line above?

File: src/core/font_renderer.js [hunk 2]
```diff
@@ -674,7 +688,9 @@ var FontRendererFactory = (function FontRendererFactoryClosure() {
     this.cmap = cmap;
     this.glyphNameMap = glyphNameMap || getGlyphsUnicode();
 
-    this.compiledGlyphs = [];
+    this.compiledGlyphs = Object.create(null);
+    this.compiledCharCodeToGlyphId = Object.create(null);
```
Comment (Snuffleupagus) [inline src/core/font_renderer.js]: Comment https://github.com/mozilla/pdf.js/pull/7106/files#r57228709 should apply here too.

**Comment by Snuffleupagus** [comment]
Updated to fix a merge conflict in `test/pdfs/.gitignore`.

**Comment by yurydelendik** [comment]
Looks good. Thank you for the patch.

/botio makeref

**Pull request merged**